bindings/go/scip/position_test.go

```diff
@@ -0,0 +1,156 @@
+package scip
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"pgregory.net/rapid"
+)
+
+// TODO: Replace with cmp.Compare go 1.21 or newer.
+func compare(i1 int, i2 int) int {
+	if i1 < i2 {
+		return -1
+	}
+	if i1 > i2 {
+		return 1
+	}
+	return 0
+}
+
+func TestUnitComparePosition(t *testing.T) {
+	positions := []Position{
+		{0, 0},
+		{0, 1},
+		{1, 0},
+		{1, 1},
+		{1, 2},
+		{2, 1},
+		{2, 2},
+	}
+
+	type testCase struct {
+		p1       Position
+		p2       Position
+		expected int
+	}
+
+	testCases := make([]testCase, len(positions)*len(positions))
+
+	// There is a total ordering on these positions, so we just test all possible combinations
+	for i1, pos1 := range positions {
+		for i2, pos2 := range positions {
+			testCases = append(testCases, testCase{pos1, pos2, compare(i1, i2)})
+		}
+	}
+
+	for _, testCase := range testCases {
+		if actual := testCase.p1.Compare(testCase.p2); actual != testCase.expected {
+			t.Errorf("unexpected result. %+v.Compare(%+v) want=%d have=%d", testCase.p1, testCase.p2, testCase.expected, actual)
+		}
+	}
+}
+
+func TestUnitIntersect(t *testing.T) {
+	r1 := Range{Position{0, 0}, Position{0, 10}}
+	r2 := Range{Position{0, 5}, Position{1, 5}}
+	r3 := Range{Position{1, 0}, Position{1, 10}}
+
+	require.Truef(t, r1.Intersects(r2), "%+v.Intersects(%+v)", r1, r2)
+	require.Truef(t, r2.Intersects(r3), "%+v.Intersects(%+v)", r2, r3)
+	require.Falsef(t, r1.Intersects(r3), "%+v.Intersects(%+v)", r1, r3)
+}
+
+func genPosition() *rapid.Generator[Position] {
+	return rapid.Custom(func(t *rapid.T) Position {
+		return Position{
+			Line:      rapid.Int32Range(0, 10).Draw(t, "Line"),
+			Character: rapid.Int32Range(0, 10).Draw(t, "Character"),
```

Comment:
Oh, I made these ranges really small to trigger a case I wanted to see fail. I think these should be `rapid.Int32Range(0, math.MaxInt32)` (Assuming `rapid.Int32Range` is `[max..min]` I couldn't find documentation)

Reply:
Or maybe `(0..100)` if we're worried we otherwise rarely get ranges that interact with each other...

Reply:
We're not doing any math, so I don't think it's worth testing the full range. Rather, more tests in a smaller range are reasonably likely to exercise all the code paths.